### PR TITLE
Added support for segmentation models to support .export()

### PR DESCRIPTION
**Pull request by BloodAxe**
* Added notebook showing how to use it
* Added a link in README
* Added a unit test
* Added necessary code changes to ensure all works smoothly

### CODE REVIEW

**Review by ofrimasad — COMMENTED** [review]
very nice work!
some comments inline

File: src/super_gradients/module_interfaces/exportable_segmentation.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+        # Hard-coded for now
+        # Will be made a parameter if we decide to support CoreML/OpenVino/TRT export in the future
+        engine = ExportTargetBackend.ONNXRUNTIME
+
+        try:
+            from pytorch_quantization import nn as quant_nn
+            from super_gradients.training.utils.quantization.calibrator import QuantizationCalibrator
+            from super_gradients.training.utils.quantization.selective_quantization_utils import SelectiveQuantizer
+
+            patch_pytorch_quantization_modules_if_needed()
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_segmentation.py:238]: No it will not. We have `import_onnx_graphsurgeon_or_install` for graphsurgeon package, but not for torch quantization yet.

File: src/super_gradients/module_interfaces/exportable_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+        :param threshold: A threshold value for converting logits to hard binary mask.
+        """
+        super().__init__()
+        self.threshold = threshold
+
+    def forward(self, predictions: Any) -> Tensor:
+        if isinstance(predictions, (list, tuple)):
+            predictions = predictions[0]
+
+        return (predictions[:, 0].sigmoid() >= self.threshold).long()
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_segmentation.py:81]: This assumes model outputs logits - (-inf,+inf) which is default output for BCE and other training objectives.

File: src/super_gradients/training/processing/processing.py [hunk 3]
```diff
@@ -730,6 +731,14 @@ def get_equivalent_photometric_module(self) -> Optional[nn.Module]:
     def resizes_image(self) -> bool:
         return True
 
+    def infer_image_input_shape(self) -> Optional[Tuple[int, int]]:
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/training/processing/processing.py:734]: There is no mistake.
The goal of the method is to help us infer the image shape (H,W) for a given model.
There is a `infer_image_shape_from_model` method that we use during export.
So based on how we trained a model this method will return a (H,W) tuple for us.
For this is goes over stored image preprocessing metadata and calls `infer_image_input_shape` on each preprocessing step.
So by checking the **output shape** of the **preprocessing** steps
 we are inferring the **input shape** of the image for the model.

File: src/super_gradients/module_interfaces/exportable_segmentation.py [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+        explicitly specified via `format` argument.
+
+        :param output: Output file name of the exported model.
+        :param confidence_threshold: (float) Confidence threshold for the exported model.
+               This parameter is used only for binary segmentation models (num_classes = 1).
+               If None, a default value of 0.5 will be used.
+        :param quantization_mode: (QuantizationMode) Sets the quantization mode for the exported model.
+            If None, the model is exported as-is without any changes to mode weights.
+            If QuantizationMode.FP16, the model is exported with weights converted to half precision.
+            If QuantizationMode.INT8, the model is exported with weights quantized to INT8. For this mode you can use calibration_loader
```
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_segmentation.py]: Fixed in ad7228f2bca8f0ba61de1bc65deec24808f6733d

File: src/super_gradients/module_interfaces/exportable_segmentation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BloodAxe) [reply src/super_gradients/module_interfaces/exportable_segmentation.py:89]: Docs updated in ad7228f2bca8f0ba61de1bc65deec24808f6733d

**Review by shaydeci — APPROVED** [review]
LGTM
